src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakPickerMobilogram.h

```diff
@@ -0,0 +1,166 @@
+#include <vector>
+
+namespace OpenMS
+{
+    /**
+    @brief The PeakPickerMobilogram finds peaks a single mobilogram.
+
+    @htmlinclude OpenMS_PeakPickerMobilogram.parameters
+
+    It uses the PeakPickerHiRes internally to find interesting seed candidates.
+    These candidates are then expanded and a right/left border of the peak is
+    searched.
+    Additionally, overlapping peaks can be removed.
+
+    */
+
+    class OPENMS_DLLAPI PeakPickerMobilogram :
+            public DefaultParamHandler
+    {
+    public:
+        //@{
+        /// Constructor
+        PeakPickerMobilogram();
+
+        /// Destructor
+        ~PeakPickerMobilogram() override {}
+        //@}
+
+        /// indices into FloatDataArrays of resulting picked mobilogram
+        enum FLOATINDICES { IDX_FWHM = 0, IDX_ABUNDANCE = 1, IDX_LEFTBORDER = 2, IDX_RIGHTBORDER = 3, SIZE_OF_FLOATINDICES };
+
+        /// Struct to hold peak positions
+        struct PeakPositions {
+            size_t left;    // Left position of the peak
+            size_t apex;    // Apex (highest point) position of the peak
+            size_t right;   // Right position of the peak
+        };
+
+        /**
+            @brief Finds peaks in a single mobilogram and annotates left/right borders
+
+            It uses a modified algorithm of the PeakPickerHiRes
+
+            This function will return a picked mobilogram
+        */
+        void pickMobilogram(const Mobilogram& mobilogram, Mobilogram& picked_mobilogram);
+
+        /**
+            @brief Finds peaks in a single mobilogram and annotates left/right borders
+
+            It uses a modified algorithm of the PeakPickerHiRes
+
+            This function will return a picked mobilogram and a smoothed mobilogram
+        */
+        void pickMobilogram(Mobilogram mobilogram, Mobilogram& picked_mobilogram, Mobilogram& smoothed_mobilogram);
+
+        /**
+          @brief Helper function to find the highest peak in a mobilogram
+
+          This function takes the integrated intensities, left widths, right widths from the peak picker and finds the highest peak in the mobilogram. The left, apex, and right positions of the highest peak are returned.
+
+          If the peak picker found no peaks (i.e. intensities is empty), the returned PeakPositions object will have it's left, apex, and right set based on the input im_size. The peak picker could fail if the input mobilogram is sparse.
+
+          @note The left, apex, and right positions are indices into the mobilogram.
+
+          @param[in] intensities A vector of doubles containing the intensities of the peaks in the mobilogram
+          @param[in] left_widths A vector of Size values containing the left widths of the peaks in the mobilogram
+          @param[in] right_widths A vector of Size values containing the right widths of the peaks in the mobilogram
+          @param[in] im_size The size/length of the mobilogram
```

Comment:
why needed?

Reply:
I added `im_size` to return the highest PeakPosition as the original input XIM for the event if the PP failed due to a sparse or single point XIM, then it will just return a peak index boundary of the original shape of the input XIM, which is done by using `im_size`.